### DISCUSSION TITLE
CURLOPT_WRITEFUNCTION - EOF?

### DISCUSSION THREAD

**Discussion by T200proX7** [curl/curl#7798] [Q&A] [answered]
Hi guys,

Im using the CURLOPT_WRITEFUNCTION option to receive data. I saw my function gets called every time a chunk is received. Unfortunately i do not receive a 0-byte length chunk as a last callback "as a signal" that EOF is reached.

In case i receive a Content-Length in the response header, i can count the bytes and compare. But in case i do not receive a Content-Length by the Server, how can i determine its the last chunk received (transfer completed)?

Thank you.

**Comment by bagder** [comment, marked as answer]
You know the response is complete when `curl_easy_perform()` returns, or when using the multi interface, when `curl_multi_info_read` says `CURLMSG_DONE`.
